src/compiler/program.ts

```diff
@@ -473,10 +592,12 @@ namespace ts {
         }
 
         const filesByName = createMap<SourceFile | undefined>();
+        let missingFilePaths: Path[];

```

Comment:
`getMissingFilePaths` promises to return a defined result -- I would make this a `ReadonlyArray<Path>` and make it `emptyArray` by default.